rust_snuba/src/processors/spans.rs

```diff
@@ -0,0 +1,406 @@
+            log::error!("Failed to serialize processed message: {}", err);
+            InvalidMessage
+        })?;
+
+        return Ok(BytesInsertBatch {
+            rows: vec![serialized],
+        });
+    }
+    Err(InvalidMessage)
+}
+
+#[derive(Debug, Deserialize)]
+struct FromSpanMessage {
+    #[serde(default)]
+    description: String,
+    duration_ms: u64,
+    event_id: Uuid,
+    exclusive_time_ms: u64,
+    #[serde(deserialize_with = "hex_to_u64")]
+    group_raw: u64,
+    is_segment: bool,
+    #[serde(deserialize_with = "hex_to_u64")]
+    parent_span_id: u64,
+    project_id: u64,
+    retention_days: u32,
+    #[serde(deserialize_with = "hex_to_u64")]
+    segment_id: u64,
+    sentry_tags: SentryTags,
+    #[serde(deserialize_with = "hex_to_u64")]
+    span_id: u64,
+    start_timestamp_ms: u64,
+    tags: BTreeMap<String, Value>,
+    trace_id: Uuid,
+}
+
+#[derive(Debug, Default, Serialize, Deserialize)]
+struct SentryTags {
+    #[serde(default)]
+    action: String,
+    #[serde(default)]
+    domain: String,
+    #[serde(default, skip_serializing)]
+    group: String,
+    #[serde(skip_serializing, rename = "http.method")]
+    http_method: Option<String>,
+    #[serde(default)]
+    module: String,
+    #[serde(default)]
+    op: String,
+    #[serde(skip_serializing)]
+    status: SpanStatus,
+    #[serde(skip_serializing)]
+    status_code: Option<u32>,
+    #[serde(default, skip_serializing)]
+    system: String,
+    #[serde(default, skip_serializing)]
+    transaction: String,
+    #[serde(skip_serializing, rename = "transaction.method")]
+    transaction_method: Option<String>,
+    #[serde(default, skip_serializing, rename = "transaction.op")]
+    transaction_op: String,
+}
+
+impl SentryTags {
+    fn to_keys(&self) -> Vec<String> {
+        vec![
+            "action".into(),
+            "domain".into(),
+            "group".into(),
+            "http.method".into(),
+            "module".into(),
+            "op".into(),
+            "status".into(),
+            "status_code".into(),
+            "system".into(),
+            "transaction".into(),
+            "transaction.method".into(),
+            "transaction.op".into(),
+        ]
+    }
+
+    fn to_values(&self) -> Vec<Value> {
+        vec![
+            self.action.clone().into(),
+            self.domain.clone().into(),
+            self.group.clone().into(),
+            self.http_method.clone().into(),
+            self.module.clone().into(),
+            self.op.clone().into(),
+            self.status.clone().as_str().into(),
+            self.status_code.unwrap_or(0).to_string().into(),
+            self.system.clone().into(),
+            self.transaction.clone().into(),
+            self.transaction_method.clone().into(),
+            self.transaction_op.clone().into(),
+        ]
+    }
+}
+
+#[derive(Debug, Default, Serialize)]
+struct Span {
+    deleted: u8,
+    description: String,
+    duration: u64,
```

Comment:
Please review all of the types here. I just double checked this one and it does not match the ClickHouse storage definition which says this is a u32. I did not review at the rest of the list.